libs/compact-map/src/Data/Compact/KeyMap.hs

```diff
@@ -508,101 +579,111 @@ union4 n combine x y = case3 emptyC1 leafF1 arrayF1 x
               case (testBit bm1 i, testBit bm2 i) of
                 (True, False) -> index arr1 (indexFromSegment bm1 i)
                 (False, True) -> index arr2 (indexFromSegment bm2 i)
-                (False, False) -> Empty -- This should be impossible 'i' is in (bm1 .|. bm2). so it must be in bm1 or bm2 or both
+                (False, False) ->
+                  -- This should be impossible 'i' is in (bm1 .|. bm2).  so it
+                  -- must be in bm1 or bm2 or both
+                  Empty
                 (True, True) ->
-                  union4
+                  unionInternal
                     (n + 1)
                     combine
                     (index arr1 (indexFromSegment bm1 i))
                     (index arr2 (indexFromSegment bm2 i))
 
 unionWithKey :: (Key -> v -> v -> v) -> KeyMap v -> KeyMap v -> KeyMap v
-unionWithKey comb x y = union4 0 comb x y
+unionWithKey = unionInternal 0
+{-# INLINE unionWithKey #-}
 
 unionWith :: (v -> v -> v) -> KeyMap v -> KeyMap v -> KeyMap v
-unionWith comb x y = union4 0 (\_k a b -> comb a b) x y
+unionWith comb = unionInternal 0 (\_k a b -> comb a b)
+{-# INLINE unionWith #-}
 
 union :: KeyMap v -> KeyMap v -> KeyMap v
-union x y = union4 0 (\_k a _b -> a) x y
+union = unionInternal 0 (\_k a _b -> a)
+{-# INLINE union #-}
 
 -- ===========================================
 -- intersection operators
 
-intersect3 :: Int -> (Key -> u -> v -> w) -> KeyMap u -> KeyMap v -> KeyMap w
-intersect3 _ _ Empty Empty = Empty
-intersect3 n combine x y = case3 Empty leafF1 arrayF1 x
+intersectInternal :: Int -> (Key -> u -> v -> w) -> KeyMap u -> KeyMap v -> KeyMap w
+intersectInternal _ _ Empty Empty = Empty
+intersectInternal n combine x y = case3 Empty leafF1 arrayF1 x
```

Comment:
intersect', or maybe use insertInternal instead of insert', so all use the same naming convention.

Reply:
Yep, that was my thinking too :) Done